src/coinbase/validator.ts

```diff
@@ -151,11 +156,119 @@ export class Validator {
     }
   }
   /**
-   * Returns the string representation of the Validator.
+   * Returns the network ID.
    *
-   * @returns The string representation of the Validator.
+   * @returns The network ID.
+   */
+  public getNetworkId(): string {
+    return this.model.network_id;
+  }
+
+  /**
+   * Returns the asset ID.
+   *
+   * @returns The asset ID.
+   */
+  public getAssetId(): string {
+    return this.model.asset_id;
+  }
+
+  /**
+   * Returns the activation epoch of the validator.
+   *
+   * @returns The activation epoch as a string.
+   */
+  public getActivationEpoch(): string {
```

Comment:
the problem with this is - these are very ethereum specific.

In future if we say support sol validators, do we still want user to do `validator. getActivationEpoch()` ?

Do we want to support a helper function of this format instead `validator. getEthereumDetails().getActivationEpoch()` ?

Reply:
I'd be more in favor of revisiting this when we cross that bridge. i.e. I don't see us exposing Solana validator staking anytime soon.